oauth2client/client.py

```diff
@@ -1222,6 +1222,43 @@ def create_scoped(self, scopes):
         """
         return self
 
+    @classmethod
+    def from_json(cls, s):
+        from oauth2client.service_account import _ServiceAccountCredentials
+        s = _from_bytes(s)
+        data = json.loads(s)
+
+        # A GoogleCredentials.from_stream return value might be a
+        # _ServiceAccountCredentials instance, so we should handle it in from_json.
+        if (data['_module'] == 'oauth2client.service_account' and
+            data['_class'] == '_ServiceAccountCredentials'):
+            return _ServiceAccountCredentials.from_json(s)
+
+        if (data.get('token_expiry') and
+                not isinstance(data['token_expiry'], datetime.datetime)):
+            try:
+                data['token_expiry'] = datetime.datetime.strptime(
+                    data['token_expiry'], EXPIRY_FORMAT)
+            except ValueError:
+                data['token_expiry'] = None
+        retval = cls(
```

Comment:
Isn't this already done on [`Credentials._to_json`][1]? If not, could it be re-purposed / upgraded to support this?

[1]: https://github.com/google/oauth2client/blob/8f8df1f5977ec76d50cb3dca9a31044cfd1fb9b4/oauth2client/client.py#L252